packages/compass-aggregations/src/components/pipeline-results-workspace/pipeline-results-header.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { css, spacing, ErrorSummary } from '@mongodb-js/compass-components';
+
+import type { RootState } from '../../modules';
+import { retryAggregation } from '../../modules/aggregation';
+
+import type { ResultsViewType } from './pipeline-results-list';
+import PipelinePagination from './pipeline-pagination';
+import PipelineResultsViewControls from './pipeline-results-view-controls';
+
+type PipelineResultsHeaderProps = {
+  className: string;
+  error?: string;
+  onRetry: () => void;
+  onChangeResultsView: (viewType: ResultsViewType) => void;
+  resultsView: ResultsViewType;
+};
+
+const controlsStyles = css({
+  display: 'flex',
+  gap: spacing[2],
+  justifyContent: 'flex-end',
+  alignItems: 'center',
+});
+
+export const PipelineResultsHeader: React.FunctionComponent<PipelineResultsHeaderProps> =
+  ({ className, error, onRetry, onChangeResultsView, resultsView }) => {
+    return (
+      <div data-testid="pipeline-results-header" className={className}>
+        <div className={controlsStyles}>
+          <PipelinePagination />
+          <PipelineResultsViewControls
+            value={resultsView}
+            onChange={onChangeResultsView}
+          />
+        </div>
+        {error && (
+          <ErrorSummary
+            dataTestId="pipeline-results-error"
+            errors={[{ message: error }]}
```

Comment:
I think the `ErrorSummary` component is useful when we need to display multiple errors, are we planning to show more than one error here? If not maybe a `Banner` with the variant `danger="danger"` will suffice?

Reply:
Just want to double check for my understanding: shouldn't ErrorSummary be the way we display errors banners everywhere in Compass? I thought that we want to unify these things across the whole app by having a single component in compass-components to use for something like that

Reply:
I see your point, having it in one place might make these a bit easier to update instead of straight using the `Banner` variant `danger`. I wasn't intending that when I moved the `ErrorSummary` to `compass-components` recently, but it makes sense and I like it. Thanks for opening that up. We can treat `WarningSummary` the same. Nothing to change! Nice